server/server.go

```diff
@@ -3593,3 +3599,42 @@ func (s *Server) updateRemoteSubscription(acc *Account, sub *subscription, delta
 
 	s.updateLeafNodes(acc, sub, delta)
 }
+
+func (s *Server) startRateLimitLogExpiration() {
+	s.mu.Lock()
+	interval := int64(time.Second)
+	s.rateLimitLoggingCh = make(chan int64, 1)
+	s.mu.Unlock()
+
+	s.startGoRoutine(func() {
+		defer s.grWG.Done()
+
+		ticker := time.NewTicker(time.Second)
+		for {
+			select {
+			case <-s.quitCh:
+				return
+			case interval = <-s.rateLimitLoggingCh:
```

Comment:
You lock above to set but not here to access.

Reply:
The lock above is to actually create the channel to change the interval. Note that so far it will just be used for tests. So here, the method below (startRateLimitLogExpiration()) will send to the channel the new interval. No lock needed.